events/manager/manager.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manager
+
+import (
+	"time"
+
+	"k8s.io/heapster/events/core"
+
+	"github.com/golang/glog"
+)
+
+type Manager interface {
+	Start()
+	Stop()
+}
+
+type realManager struct {
+	source    core.EventSource
+	sink      core.EventSink
+	frequency time.Duration
+	stopChan  chan struct{}
+}
+
+func NewManager(source core.EventSource, sink core.EventSink, frequency time.Duration) (Manager, error) {
+	manager := realManager{
+		source:    source,
+		sink:      sink,
+		frequency: frequency,
+		stopChan:  make(chan struct{}),
+	}
+
+	return &manager, nil
+}
+
+func (rm *realManager) Start() {
+	go rm.Housekeep()
+}
+
+func (rm *realManager) Stop() {
+	rm.stopChan <- struct{}{}
+}
+
+func (rm *realManager) Housekeep() {
+	for {
+		// Always try to get the newest metrics
+		now := time.Now()
+		start := now.Truncate(rm.frequency)
+		end := start.Add(rm.frequency)
+		timeToNextSync := end.Sub(now)
```

Comment:
Please add a comment why you're doing it this way, instead of using time.After(rm.frequency). My guess is that you want to do all polls in the same time.